.github/workflows/run-all-e2e-tests.yaml

```diff
@@ -0,0 +1,57 @@
+# Runs Azure tests when pull request opened, repopened or synchronized
+name: E2E Tests - Run all tests on PR
```

Comment:
How does this interact with https://github.com/mesosphere/konvoy-image-builder/pull/717, can we still trigger individual provider's tests?

Reply:
with `workflow_dispatch` we can still trigger individual provider's test manually from GHA UI. However they might not reflect on the `Status check` list of the PR. 
I can port logic in https://github.com/mesosphere/konvoy-image-builder/pull/717 to this PR.

Reply:
ported configuration for executing tests on a given provider